src/components/common/Directors.vue

```diff
@@ -831,98 +786,105 @@ export default class Directors extends Mixins(CommonMixin, DateMixin, DirectorMi
-
-              // ensure there is complete address data including missing/blank fields
-              directors[i].deliveryAddress = this.formatAddress(directors[i].deliveryAddress)
-              if (directors[i].mailingAddress) {
-                directors[i].mailingAddress = this.formatAddress(directors[i].mailingAddress)
-              }
+      await axios.get(url).then(response => {
+        if (response?.data?.directors) {
+          const directors = response.data.directors as DirectorIF[]
+
+          this.original = directors.sort(this.fieldSorter(['lastName', 'firstName', 'middleName']))
+
+          this.original.forEach((director, i) => {
+            director.id = i + 1
+            director.isFeeApplied = (director.isFeeApplied !== undefined) ? director.isFeeApplied : false
+            director.isDirectorActionable = (director.cessationDate === null)
+            director.actions = []
+
+            // if there is no officer middle initial field, add it with blank data
+            if (!director.officer.hasOwnProperty('middleInitial')) director.officer.middleInitial = ''
+
+            // save previous officer name data for COLIN to use when updating record
+            director.officer.prevFirstName = director.officer.firstName
+            director.officer.prevLastName = director.officer.lastName
+            director.officer.prevMiddleInitial = director.officer.middleInitial
+
+            // save complete address data including missing/blank fields
+            director.deliveryAddress = this.formatAddress(director.deliveryAddress)
+            if (director.mailingAddress) {
+              director.mailingAddress = this.formatAddress(director.mailingAddress)
             }
+          })
+        } else {
+          throw new Error('Invalid directors list')
+        }
+      }).catch(error => {
+        // eslint-disable-next-line no-console
+        console.log('fetchDirectors() error =', error)
+        // re-throw error
+        throw error
+      })
+    }
+  }
 
-            // save to component data now that extra attributes are added
-            if (!getOrigOnly) this.directors = directors
-
-            // save version of directors before changes (deep copy, not reference)
-            this.directorsOriginal = cloneDeep(directors)
-          } else {
-            // eslint-disable-next-line no-console
-            console.log('getDirectors() invalid response =', response)
-          }
-        })
-        .catch(error => {
-          // eslint-disable-next-line no-console
-          console.log('getDirectors() error =', error)
-
-          // emit event to parent to display Fetch Error Dialog
-          this.$root.$emit('fetch-error-event')
-        })
+  /** Called externally to _asynchronously_ get the original directors on As Of Date. */
+  // FUTURE: this logic should be in the parent component
+  public async getOrigDirectors (asOfDate: string, updateWorkingDirectors: boolean): Promise<void> {
```

Comment:
This is very similar to a method in OfficeAddresses (ie, consistency of architecture).